server/ws/websockets.go

```diff
@@ -409,11 +410,14 @@ func (ws *Server) BroadcastBlockChange(workspaceID string, block model.Block) {
 				mlog.Stringer("remoteAddr", listener.RemoteAddr()),
 			)
 
+			ws.broadcastMutex.Lock()
```

Comment:
This is locking the whole websocket, instead of locking the listener individually, I think this can enter like this in the repo but we need a more fine-grained locks to avoid cotention here.

Reply:
I see. We should prevent concurrent writes on a single connection but allow them on different connections.

Let me update that.

Reply:
Done.